numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -271,7 +271,7 @@ gentype_multiply(PyObject *m1, PyObject *m2)
             ((Py_TYPE(m2)->tp_as_number == NULL) ||
              (Py_TYPE(m2)->tp_as_number->nb_multiply == NULL))) {
         /* Try to convert m1 to an int and try sequence repeat */
-        repeat = PyArray_PyIntAsIntp(m1);
+        repeat = PyNumber_AsSsize_t(m1, PyExc_OverflowError);
         if (error_converting(repeat)) {
             return NULL;
         }
```

Comment:
Simply using the following for this whole chunk seems to work correctly:
```
        Py_INCREF(Py_NotImplemented);
        return Py_NotImplemented;
```

Reply:
The only test that fails is the one that fails in this patch, because the warning needs silencing.

Reply:
Yes I agree that returning NotImplemented is better.